### PR TITLE
Improvement of Gaussian Troubleshooting

### PR DESCRIPTION
**Pull request by calvinp0**
Currently our attempt at troubleshooting Gaussian is not performing the best it can. We do not do combination of troubleshoots. This PR attempts to fix that

Need input on how to do the integral algorithm change better

### CODE REVIEW

File: arc/job/trsh.py [hunk 1]
```diff
@@ -851,48 +851,64 @@ def trsh_ess_job(label: str,
...
+            ess_trsh_methods.append('checkfile=None')
             remove_checkfile = True
         elif 'InternalCoordinateError' in job_status['keywords'] \
                 and 'cartesian' not in ess_trsh_methods and job_type == 'opt':
             # try both cartesian and nosymm
             logger.info(f'Troubleshooting {job_type} job in {software} for {label} using opt=cartesian with nosyym')
             ess_trsh_methods.append('cartesian')
             trsh_keyword = 'opt=(cartesian,nosymm)'
+            if 'CheckFile' in job_status['keywords'] or 'checkfile=None' in ess_trsh_methods:
+                remove_checkfile = True
```
Comment (alongd) [inline arc/job/trsh.py:869]: This approach only solves combinations for checkfile specifically. What if we need to combine two other keywords? Would you like to keep around a list of trsh keywords as a Job argument, and append to it as needed?

File: arc/scheduler.py [hunk 2]
```diff
@@ -637,6 +637,8 @@ def schedule_jobs(self):
                             if successful_server_termination \
                                     and (job.directed_scan_type is None or job.directed_scan_type == 'ess'):
                                 self.check_scan_job(label=label, job=job)
+                            elif successful_server_termination and job.job_status[1]['status'] == 'errored':
```
Comment (alongd) [inline arc/scheduler.py:640]: should this be more generic than just for scan jobs?

File: arc/species/species.py [hunk 3]
```diff
@@ -561,9 +561,8 @@ def number_of_heavy_atoms(self) -> int:
                 self._number_of_heavy_atoms = len([symbol for symbol in self.get_xyz()['symbols'] if symbol != 'H'])
             elif self.is_ts:
                 for ts_guess in self.ts_guesses:
-                    if ts_guess.xyz is not None:
-                        self._number_of_heavy_atoms = \
-                            len([line for line in ts_guess.xyz.splitlines() if line.split()[0] != 'H'])
+                    if ts_guess.get_xyz() is not None:
```
Comment (alongd) [inline arc/species/species.py:564]: Can you add a test?

File: arc/job/adapters/gaussian_test.py [hunk 4]
```diff
@@ -154,7 +154,7 @@ def test_write_input_file(self):
 %mem=14336mb
 %NProcShared=8
 
-#P opt=(calcfc) cbs-qb3   IOp(2/9=2000) IOp(1/12=5,3/44=0) scf=xqc  
+#P opt=(calcfc) cbs-qb3   IOp(2/9=2000) IOp(1/12=5,3/44=0) scf=xqc
```
Comment (alongd) [inline arc/job/adapters/gaussian_test.py:200]: Can you undo the changes to this file (I think that only some spaces were added)

File: arc/job/adapters/gaussian.py [hunk 5]
```diff
@@ -48,7 +48,7 @@
 %%mem=${memory}mb
 %%NProcShared=${cpus}
 
-#P ${job_type_1} ${restricted}${method}${slash_1}${basis}${slash_2}${auxiliary_basis} ${job_type_2} ${fine} IOp(2/9=2000) ${keywords} ${dispersion}
+#P ${job_type_1} ${restricted}${method}${slash_1}${basis}${slash_2}${auxiliary_basis} ${job_type_2} ${fine} IOp(2/9=2000) ${keywords} ${dispersion} ${trsh}
```
Comment (alongd) [inline arc/job/adapters/gaussian.py:51]: `${trsh}` not used in this commit, is it needed?

File: arc/job/adapters/gaussian_test.py [hunk 6]
```diff
@@ -27,110 +27,153 @@ def setUpClass(cls):
...
-                                    job_type='composite',
-                                    level=Level(method='cbs-qb3-paraskevas'),
-                                    project='test',
-                                    project_directory=os.path.join(ARC_PATH, 'arc', 'testing', 'test_GaussianAdapter'),
-                                    species=[ARCSpecies(label='spc1', xyz=['O 0 0 1'])],
-                                    testing=True,
-                                    args={'keyword': {'general': 'IOp(1/12=5,3/44=0)'}},
-                                    )
-        cls.job_9 = GaussianAdapter(execution_type='local',
+        # cls.job_1 = GaussianAdapter(execution_type='incore',
```
Comment (alongd) [inline arc/job/adapters/gaussian_test.py:30]: these tests were commented out, is it just for debugging? Can you return them back?

File: arc/job/adapters/gaussian_test.py [hunk 7]
```diff
@@ -27,110 +27,153 @@ def setUpClass(cls):
...
+                            level=Level(method='wb97xd'),
+                            fine=True,
                             project='test',
                             project_directory=os.path.join(ARC_PATH, 'arc', 'testing', 'test_GaussianAdapter'),
                             species=[ARCSpecies(label='anion', xyz=['O 0 0 1'], charge=-1, is_ts=False)],
                             testing=True,
+                            args={'trsh': {'trsh': 'Acc2E=14'}},
                             )
 
+        # Need to setup trsh for job_11
```
Comment (alongd) [inline arc/job/adapters/gaussian_test.py:144]: unclear comment

File: arc/job/adapters/gaussian_test.py [hunk 8]
```diff
@@ -27,110 +27,153 @@ def setUpClass(cls):
...
+                # Test Gaussian
```
Comment (alongd) [inline arc/job/adapters/gaussian_test.py:145]: misaligned comment

File: arc/job/adapters/gaussian_test.py [hunk 9]
```diff
@@ -246,11 +289,11 @@ def test_write_input_file(self):
...
 anion
 
--1 2
-O       0.00000000    0.00000000    1.00000000
+    -1 2
+    O       0.00000000    0.00000000    1.00000000
 
 
-"""
+    """
```
Comment (alongd) [inline arc/job/adapters/gaussian_test.py:296]: are these changes needed?

File: arc/job/adapters/gaussian_test.py [hunk 10]
```diff
@@ -319,6 +362,49 @@ def test_set_files_for_pipe(self):
     def test_gaussian_def2tzvp(self):
         """Test a Gaussian job using def2-tzvp"""
         self.assertEqual(self.job_9.level.basis.lower(), 'def2tzvp')
+    
+    def test_trsh_write_input_file(self):
+        """Test writing a trsh input file
+        1. Test with getting Acc2E14 as the trsh method and thus changing the input file integral algorithm
+        2. Test with getting Acc2E14 as the trsh method but also checkfile=None in ess_trsh_methods, thus it will have both changes in the input file
+
```
Comment (alongd) [inline arc/job/adapters/gaussian_test.py:370]: please remove extra blank line

File: arc/job/adapters/gaussian_test.py [hunk 11]
```diff
@@ -319,6 +362,49 @@ def test_set_files_for_pipe(self):
...
+anion
+
+-1 2
+O       0.00000000    0.00000000    1.00000000
+
+
+"""
+        self.assertEqual(content_10, job_10_expected_input_file)
+
```
Comment (alongd) [inline arc/job/adapters/gaussian_test.py:389]: please remove extra blank line

File: arc/job/trsh_test.py [hunk 12]
```diff
@@ -283,7 +283,7 @@ def test_trsh_ess_job(self):
         self.assertTrue(remove_checkfile)
         self.assertEqual(software, 'gaussian')
         self.assertEqual(memory, 16)
-        self.assertFalse(couldnt_trsh)
+        self.assertFalse(couldnt_trsh)
```
Comment (alongd) [inline arc/job/trsh_test.py:286]: was this space addition done on purpose?

**Review by alongd — COMMENTED** [review]
Thanks @calvinp0 ! I added some comments, please look at the tests

File: arc/job/adapters/gaussian.py (hunk 5 above)
Comment (calvinp0) [reply arc/job/adapters/gaussian.py:51]: I believe it is used

```
        input_dict['trsh'] = self.args['trsh']['trsh'] if 'trsh' in self.args['trsh'] else ''
        input_dict['xyz'] = xyz_to_str(self.xyz)
        if 'Acc2E=14' in input_dict['trsh']:
            input_dict['trsh'] = input_dict['trsh'].replace('Acc2E=14', '')
        if 'checkfile=None' in input_dict['trsh']:
            input_dict['trsh'] = input_dict['trsh'].replace('checkfile=None', '')
        integral_algorithm = 'Acc2E=14' if (self.args.get('trsh') and self.args['trsh'].get('trsh') and 'Acc2E=14' in self.args['trsh']['trsh']) else 'Acc2E=12'
```

File: arc/job/trsh.py (hunk 1 above)
Comment (calvinp0) [reply arc/job/trsh.py:869]: I was trying to build in slowly combinations to be honest. I also need to figure out which combinations will not work with each other
